src/zc/recipe/testrunner/tests.py

```diff
@@ -12,57 +12,166 @@
 #
 ##############################################################################
 
+from zc.buildout.testing import mkdir, write, system
+
 import doctest
+import os
 import re
+import shutil
+import tempfile
 import unittest
-
 import zc.buildout.testing
-import doctest
 import zope.testing.renormalizing
 
 
+SETUP_PY = """from setuptools import setup
+setup(name = "bugfix1")
+"""
```

Comment:
Usually multiline strings are more readable if you insert a line break after the opening `"""`:

```python
SETUP_PY = """\
from setuptools import setup
setup(name="bugfix1")
"""
```

If you strongly dislike backslashes, another option is to `.lstrip('\n')`:


```python
SETUP_PY = """
from setuptools import setup
setup(name="bugfix1")
""".lstrip('\n')
```

When the multiline string literals are very long, or are defined in some nested scope, they can be made more readable with the help of textwrap.dedent:

```python
SETUP_PY = textwrap.dedent("""\
    from setuptools import setup
    setup(name="bugfix1")
""")
```

I do not think this is necessary here and am merely mentioning it for completeness.